### PR TITLE
[Feature]: pagination for source http

### PR DESCRIPTION
**Pull request by davama**
## Pre-Request Checklist

- [ ] Passes rubocop code analysis (try `rubocop --auto-correct`)
- [ ] Tests added or adapted (try `rake test`)
- [x] Changes are reflected in the documentation
- [x] User-visible changes appended to [CHANGELOG.md](/CHANGELOG.md)

## Description

The burden of getting ALL data should not fall on the api but on the client.

We have several applications that use pagination and thought it was time for oxidized to have it too.

Basically adds two new settings to `source.http`.
See documentation for details.

Does not break existing config settings who do not have these settings enabled.

Thanks

**EDIT**

forgot to mention that I am not a ruby dev, just a IT enthusiast. So any ruby code syntax suggestions are welcomed

### CODE REVIEW

**Comment by davama** [comment]
Will try to fix the CI checks

**Comment by davama** [comment]
ok
i've done all the code related changes.

not sure what to make of the `codecov/patch` and `codecov/project`

as far as the module, it works as described in the first comment

Please let me know what I can do to make the checks pass.

Thanks

**Comment by davama** [comment]
all checks passed! this is great!!

**Comment by davama** [comment]
This is ready to be merged 👍
